### PR TITLE
Refactor Pod models and move strings to constants

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Break the `GetPodSpec` method into smaller pieces and move repeated strings into a constant.

## Type of change

*Please select one of the options below.*

- Other (refactoring)

## Discussion

We should continue moving variable into constants that can be used everywhere instead of using magic strings, e.g. the container images.

## Testing

Local testing + unit tests.

## Documentation

-

## Follow-up

-

### CODE REVIEW

**Review by jleach4 — APPROVED** [review]
Small nit, but maybe add a comment to the new functions.

**Review by saintstack — APPROVED** [review]
The refactor of pod models is hard to follow. We have good coverage in here? Stuff will break if we put the containers together wrong? If so, +1.  The renames using constants look good.

**Review by ammolitor — APPROVED** [review]
